### ISSUE TITLE
reset env with default theme is not working

### ISSUE THREAD

**Issue by SungbinChoi** [Unity-Technologies/obstacle-tower-env#98]
When I create an UnityEnvironment and then reset it with specific default theme option, first time it's done as I specified. But then I reset env with different default theme option, env theme is not changed.

e.g.,  

// Create env
env = UnityEnvironment(  ...) 

// Reset with theme '1'  
reset_params = {}
reset_params['default-theme'] = 1  
reset_params['visual-theme'] = 0
info = env.reset(config=reset_params,
                          train_mode=True)[brain_name]      
=>   env default-theme is reset to 1

// Reset with theme '2'   
reset_params['default-theme'] = 2  
reset_params['visual-theme'] = 0
info = env.reset(config=reset_params,
                          train_mode=True)[brain_name]      
=>   env default-theme is still 1
       although INFO message said env reset with 'default-theme -> 2'

**Comment by awjuliani** [comment]
Hello @SungbinChoi 

This is indeed a bug. We will address it in a future release. Right now a workaround is to reset by changing the seed, and then change it back and that will allow the theme to change.

**Comment by vkakerbeck** [comment]
Hi,

I have a similar issue. I am using a curriculum and some of the parameters, including the visual theme , don't seem to reset correctly (visual theme when reset for the second time, other fail also on the first time). I've noticed it first when the agent would walk past the 5th floor without picking up a key even though the lesson had allowed_rooms set to 1 or 2. When starting the curriculum in one of the more advanced lessons it correctly sets the harder parameters and the agent picks up a key before progressing. But when starting from the first lesson some of the parameters aren't reset while going through the curriculum. 

After a couple of test the parameters where I noticed this are allowed_rooms, lighting-type and visual theme. As a test if the problem is caused by something else I also tested starting-floor, seed and agent-perspective but these parameters seem to work. 

The hard thing about figuring this out is that when I print out the academy parameters they are exactly how I specified them. The only way I can tell that something isn't right is that there are no keys picked up or by looking at performance differences between running through the curriculum to lesson x compared to setting the reset parameters of lesson x right away.

Could this be part of the bug or is there some way I can fix this?
